src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -623,6 +639,60 @@ LIMIT 1
         return result;
     }
 
+    private async Task<Dictionary<long, GatewayModel.EntityMetadataCollection>> GetExplicitMetadata(Dictionary<long, List<string>> entityMetadataKeys, GatewayModel.LedgerState ledgerState,
+        CancellationToken token)
+    {
+        var entityIds = new List<long>();
+        var metadataKeys = new List<string>();
+
+        foreach (var (entityId, keys) in entityMetadataKeys)
+        {
+            foreach (var key in keys)
+            {
+                entityIds.Add(entityId);
+                metadataKeys.Add(key);
+            }
+        }
+
+        var metadataHistory = await _dbContext.EntityMetadataHistory
+            .FromSqlInterpolated(@$"
+WITH variables (entity_id, metadata_key) AS (
```

Comment:
I guess we want some deterministic order of results returned from that SQL and later endpoint. Would probably make sense to order by `entity_id` then by `keys`?

Reply:
Not here.

In general we care about order of things extracted from database where data is paginated and/or should be ordered by "most recently affected". Then obviously we must order it. However we don't really care about order of responses generated for HTTP requests, i.e.: `/entity/details {"addresses":["a", "b"], "explicit_metadata":["x", "y"]` - we don't necessarily have to preserve order of mentioned addresses or explicit metadata.